src/T2DMap.cpp

```diff
@@ -672,13 +672,503 @@ bool T2DMap::sizeFontToFitTextInRect( QFont & font, const QRectF & boundaryRect,
+            brush.setStyle(Qt::Dense4Pattern);
+        } else {
+            pen.setWidthF(mRoomWidth * rSize * outerStubDoorPenThicknessFactor);
+            innerPen.setWidthF(mRoomWidth * rSize * innerStubDoorPenThicknessFactor);
+            brush.setStyle(Qt::DiagCrossPattern);
+        }
+        painter.setBrush(brush);
+        painter.setPen(pen);
+        painter.drawPolygon(poly_out_left);
+        painter.drawPolygon(poly_out_right);
+        if (isDoor) {
+            painter.setPen(innerPen);
+            painter.setBrush(innerBrush);
+            painter.drawPolygon(poly_out_left);
+            painter.drawPolygon(poly_out_right);
+        }
+    }
+
+    painter.restore();
+    if (!pArea->gridMode) {
+        QMapIterator<int, QPoint> it(mAreaExitsList);
+        while (it.hasNext()) {
+            it.next();
+            QPoint P = it.value();
+            int rx = P.x();
+            int ry = P.y();
+
+            QRectF dr = QRectF(rx, ry, mRoomWidth * rSize, mRoomHeight * rSize);
+
+            // clang-format off
+            if (((mPick || picked)
+                 && mPHighlight.x() >= (dr.x() - mRoomWidth / 3.0)
+                 && mPHighlight.x() <= (dr.x() + mRoomWidth / 3.0)
+                 && mPHighlight.y() >= (dr.y() - mRoomHeight / 3.0)
+                 && mPHighlight.y() <= (dr.y() + mRoomHeight / 3.0))
+                && mStartSpeedWalk) {
+
+                // clang-format on
+                mStartSpeedWalk = false;
+                // This draws a red circle around the out of area exit that
+                // was choosen as the target for the speedwalk, but it is
+                // only shown for one paintEvent call and it is not obvious
+                // that it is useful, note that there is similar code for a
+                // room being clicked on that is WITHIN the area, that is
+                // above this point in the source code:
+                float roomRadius = (0.8 * mRoomWidth) / 2.0;
+                QPointF roomCenter = QPointF(rx, ry);
+                QRadialGradient gradient(roomCenter, roomRadius);
+                gradient.setColorAt(0.95, QColor(255, 0, 0, 150));
+                gradient.setColorAt(0.80, QColor(150, 100, 100, 150));
+                gradient.setColorAt(0.799, QColor(150, 100, 100, 100));
+                gradient.setColorAt(0.7, QColor(255, 0, 0, 200));
+                gradient.setColorAt(0, Qt::white);
+                QPen transparentPen(Qt::transparent);
+                QPainterPath myPath;
+                painter.setBrush(gradient);
+                painter.setPen(transparentPen);
+                myPath.addEllipse(roomCenter, roomRadius, roomRadius);
+                painter.drawPath(myPath);
+
+                mPick = false;
+                mTarget = it.key();
+                if (mpMap->mpRoomDB->getRoom(mTarget)) {
+                    mpMap->mTargetID = mTarget;
+                    if (mpMap->findPath(playerRoomId, mpMap->mTargetID)) {
```

Comment:
Same

Reply:
Similar as before - but this is because the *exit* from a room that goes to another area has been found to be close enough to the mouse double-click to be the destination for a speed-walk - also that exit will be drawn once (in this paintEvent only) with a red circle drawn on top of it...